From foo@baz Tue Apr  9 12:12:43 2002
To: Greg KH <greg@kroah.com>
Date: 02 Mar 2005 15:03:15 -08:00
From: khali@linux-fr.org
Subject: [PATCH] Add class definition to the elektor bus driver

Hi Frank, all,

> > Which bus driver are you using? It obviously lacks class declaration,
> > so the correct fix is to add the class there.
>
> The modules that are loading are (in reverse order):
>      adm1031
>      ad5321
>      mic184
>      pca9540
>      i2c_sensor
>      i2c_elektor
>      i2c_algo_pcf
>      i2c_core
>
> So I believe what you are asking for is the i2c_elektor driver for the
> PCF8584 ISA to I2C chip.

Correct, I just checked and this one actually lacks its class. Patch
follows.

This patch adds a class definition to the elektor i2c bus driver.
Without this definition, hardware monitoring chips located on such
busses cannot possibly be driven.

Signed-off-by: Jean Delvare <khali@linux-fr.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


diff -Nru a/drivers/i2c/busses/i2c-elektor.c b/drivers/i2c/busses/i2c-elektor.c
--- a/drivers/i2c/busses/i2c-elektor.c	2005-03-02 16:31:53 -08:00
+++ b/drivers/i2c/busses/i2c-elektor.c	2005-03-02 16:31:53 -08:00
@@ -183,6 +183,7 @@
 
 static struct i2c_adapter pcf_isa_ops = {
 	.owner		= THIS_MODULE,
+	.class		= I2C_CLASS_HWMON,
 	.id		= I2C_HW_P_ELEK,
 	.algo_data	= &pcf_isa_data,
 	.name		= "PCF8584 ISA adapter",
